frontend/public/components/dashboard/dashboards-page/cluster-dashboard/utilization-card.tsx

```diff
@@ -228,30 +235,46 @@ const UtilizationCardNodeFilter: React.FC<UtilizationCardNodeFilterProps> = ({
     }
     return indexA - indexB;
   });
-  const onToggle = (_event, open: boolean): void => setIsOpen(open);
+  // const onToggle = (_event, open: boolean): void => setIsOpen(open);
 
-  const selectedNodesUpdated = selectedNodes.map((item) =>
-    item === 'master' ? 'control plane' : item,
+  // const selectedNodesUpdated = selectedNodes.map((item) =>
+  //   item === 'master' ? 'control plane' : item,
+  // );
+
+  const selectOptions = sortedMCPs.map((mcp) => (
+    <SelectOption
+      hasCheckbox
+      key={mcp.metadata.name}
+      value={mcp.metadata.name === 'master' ? 'control plane' : mcp.metadata.name}
+      isSelected={selectedNodes.includes(mcp.metadata.name)}
+    >
+      {mcp.metadata.name}
+    </SelectOption>
+  ));
+
+  const toggle = (toggleRef: React.Ref<MenuToggleElement>) => (
+    <MenuToggle ref={toggleRef} onClick={(open) => setIsOpen(open)} variant="plainText">
+      {t('public~Filter by Node type')}
+      {selectedNodes.length > 0 && (
+        <Badge className="pf-v5-u-ml-sm" isRead>
+          {selectedNodes.length}
+        </Badge>
+      )}
+    </MenuToggle>
   );
 
   return (
-    <SelectDeprecated
-      variant={SelectVariantDeprecated.checkbox}
+    <Select
+      role="menu"
       aria-label={t('public~Filter by Node type')}
-      onToggle={onToggle}
       onSelect={onNodeSelect}
-      selections={selectedNodesUpdated}
       isOpen={isOpen}
-      placeholderText={t('public~Filter by Node type')}
-      isPlain
+      selected={selectedNodes}
+      onOpenChange={(onOpenChange) => setIsOpen(onOpenChange)}
```

Comment:
Nit: naming.  We're not consistent within the code base, but there is at least one other example that uses `open`.
```suggestion
      onOpenChange={(open) => setIsOpen(open)}
```